Logging: Allow toggle

dseptem opened the pull request:
Reworked the script based on your comments.
See commits for details.

dseptem committed:
on_command feature added

code is elegant now, on_command is off by default but can be turned on
to set enable logging on command in a channel per channel basis
on_command requires to be used on a channel and only acknowledged the
bot's owner
on_command usage example included for ".help logging"

dseptem committed:
minor typo fix

dseptem committed:
Fixes according to suggestions

-renamed from on_command to allow_toggle
-changed from require_owner to require_admin
-consistent usage of module prefix
-changed example to ".log start"
-renamed from '.logging start/stop' to '.log start/stop'
-remade write_log if condition according to suggestion

maxpowa commented on sopel_modules/chanlogs2/chanlogs2.py:
```diff
@@ -49,6 +51,7 @@ def configure(config):
         config.chanlogs2.configure_setting('pg_connection', 'PostgreSQL connection string, see http://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING for more info')
     else:
         config.chanlogs2.configure_setting('logdir', 'Log storage directory')
+        config.chanlogs2.configure_setting('allow_toggle', "Start and stop logging on an admin's command")
```
I just realized this will only work if postgres is not used. This should probably be moved up to line 49.

dseptem replied:
Fixed.

maxpowa commented on sopel_modules/chanlogs2/chanlogs2.py:
```diff
@@ -149,6 +152,24 @@ def redirect_topic(bot, trigger):
+            bot.reply('Logging stopped for this channel')
+        else:
+            bot.reply("Please, use '.log start' or '.log stop'")
+    else:
```
This is another case where you could return early to avoid wrapping most of the function in an if/else.

dseptem replied:
Fixed, I think.

maxpowa commented:
My comments on the previous one about `require_op` instead of `require_admin` were from the perspective of having the bot as a 'public service', where anyone can invite it and the logs available from a web viewer. If channel ops didn't want the bot to log their channel they could turn it off themselves rather than needing a bot admin to do it for them.

dseptem commented:
That's beyond my scope.
My work was intended for personal usage, not public usage. Of course, you're free to add those features if you wish to do so.

dseptem committed:
more fixes

fixed allow_toggle being skipped if postgres is enabled
fixed the big if wrapping the logging_command function
